packages/core/src/resolve.ts

```diff
@@ -313,6 +319,22 @@ export async function resolveDocument(opts: {
       if (hasRef(refStack.prev, ref)) {
         throw new Error('Self-referencing circular pointer');
       }
+
+      if (isAnchor(ref.$ref)) {
+        // Wait for all anchors in the document to be collected firstly.
+        await delay();
```

Comment:
This is a bit dangerous and funky 😄 

What you are doing here is waiting for the next event loop iteration, I would rename this delay function appropriately. 

There are edge-cases I am worried about but let me think about it.

Reply:
Yeah,  I agree. However, I've tried to isolate this behaviour for anchors only.  
Please let me know if you can think of a better option.

Reply:
Renamed into `nextTick`.